firebase-firestore/ktx/src/androidTest/java/com/google/firebase/firestore/NestedMapEncoderIntegrationTest.kt

```diff
@@ -0,0 +1,137 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore
+
+import com.google.firebase.firestore.ktx.serialization.setData
+import com.google.firebase.firestore.testutil.IntegrationTestUtil.Companion.testCollection
+import com.google.firebase.firestore.testutil.IntegrationTestUtil.Companion.waitFor
+import kotlinx.serialization.Serializable
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class NestedMapEncoderIntegrationTest {
+
+    @Test
+    fun serializationSetDataMethodIsTheSameAsPOJOSetOnPlainObject() {
+
+        @Serializable data class Project(val name: String? = null, val owner: String? = null)
+
+        val docRefKotlin = testCollection("ktx").document("123")
+        val docRefPOJO = testCollection("pojo").document("456")
+        val projectList = listOf(Project(), Project("x"), Project("x", "y"), Project(null, null))
+
+        for (project in projectList) {
+            docRefKotlin.setData(project)
```

Comment:
Are you set on using the additional methods instead of the module based approach?

Reply:
After I have a working `FirestoreRegistrar` and `getcomponent()`  , I will wrap this `setData()` method inside of the current API. The goal is to not expose this `setData()` to customer.